### PR TITLE
Fix AccessSummaryAnalysis to handle @convention(block) in nested noescape closures.

### PR DESCRIPTION
**Pull request by atrick**
This analysis uses a whitelist so we can be sure not do miss any diagnostics. Naturally there are some very unusual cases that weren't part of that original whitelist. As I've already pointed out, these kind of assumptions should be ratified in SILVerifier, but that will need to come in a subsequent commit.

### CODE REVIEW

**Comment by atrick** [comment]
@swift-ci test.

**Comment by devincoughlin** [comment]
Thanks Andy! This looks great.

File: lib/SILOptimizer/Analysis/AccessSummaryAnalysis.cpp [hunk 1]
```diff
@@ -129,24 +131,47 @@ void AccessSummaryAnalysis::processArgument(FunctionInfo *info,
 /// only ultimately used by an apply, a try_apply or as an argument (but not
 /// the called function) in a partial_apply.
 /// TODO: This really should be checked in the SILVerifier.
-static bool isExpectedUseOfNoEscapePartialApply(SILInstruction *user) {
-  if (!user)
-    return true;
+static bool hasExpectedUsesOfNoEscapePartialApply(SILInstruction *closure) {
+  for (Operand *use : closure->getUses()) {
+    SILInstruction *user = use->getUser();
```
Comment (rudkx) [inline lib/SILOptimizer/Analysis/AccessSummaryAnalysis.cpp]: This only ever looks at the first use since all the switch cases return.

**Comment by atrick** [comment]
Thanks. Refactor error.

File: lib/SILOptimizer/Analysis/AccessSummaryAnalysis.cpp (hunk 1 above)
Comment (atrick) [inline lib/SILOptimizer/Analysis/AccessSummaryAnalysis.cpp]: Fixed in https://github.com/apple/swift/pull/10503

**Pull request merged**